jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/PathRequestContent.java

```diff
@@ -26,7 +26,7 @@
  * {@link ByteBufferPool} from which {@code ByteBuffer}s will be acquired
  * to read from the {@code Path}.</p>
  */
-public class PathRequestContent extends PathContentSource implements Request.Content
+public class PathRequestContent extends ByteChannelContentSource.PathContentSource implements Request.Content
```

Comment:
This class should still inherit from `io.content.PathContentSource`, which itself should inherit from the new `ByteChannelContentSource`.

Reply:
@sbordet I had initially deferred these changes to 12.1, but @lorban asked for them to be in this PR.  I will back the substantive changes out and simply deprecate the methods that will change in 12.1.x